bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/dto/LightState.java

```diff
@@ -51,15 +51,46 @@ public class LightState {
 
     public @Nullable Integer transitiontime;
 
+    private static enum ColorMode {
+        CT,
+        HS,
+        XY
+    }
+
+    private @Nullable ColorMode colorModeFrom(@Nullable String value) {
+        if (value != null) {
+            try {
+                return ColorMode.valueOf(value.toUpperCase());
+            } catch (IllegalArgumentException e) {
+                // fall through
+            }
+        }
+        return null;
+    }
+
+    private boolean compareXYWithTolerance(double @Nullable [] a, double @Nullable [] b, double tolerance) {
+        return a == null || b == null || (a.length > 1) && (b.length > 1) && (Math.abs(a[0] - b[0]) < tolerance)
+                && (Math.abs(a[1] - b[1]) < tolerance);
```

Comment:
```suggestion
        return a == null || b == null || ((a.length > 1) && (b.length > 1) && (Math.abs(a[0] - b[0]) < tolerance)
                && (Math.abs(a[1] - b[1]) < tolerance));
```

Even though it works because `&&` is higher level than `||` using brackets makes it more clear.